onnxruntime/core/framework/provider_bridge_ort.cc

```diff
@@ -49,487 +76,553 @@ struct Provider_IDeviceAllocator_Impl : Provider_IDeviceAllocator {
 
-    return outer_->Provider_Compile(provider_fused_nodes, node_compute_funcs);
-  }
+  // Provider_NodeArg
+  const std::string& Provider_NodeArg__Name(const Provider_NodeArg* p) noexcept override { return reinterpret_cast<const NodeArg*>(p)->Name(); }
+  const ONNX_NAMESPACE::Provider_TensorShapeProto* Provider_NodeArg__Shape(const Provider_NodeArg* p) override { return reinterpret_cast<const ONNX_NAMESPACE::Provider_TensorShapeProto*>(reinterpret_cast<const NodeArg*>(p)->Shape()); }
+  ONNX_NAMESPACE::DataType Provider_NodeArg__Type(const Provider_NodeArg* p) noexcept override { return reinterpret_cast<const NodeArg*>(p)->Type(); }
+  const Provider_NodeArgInfo& Provider_NodeArg__ToProto(const Provider_NodeArg* p) noexcept override { return *reinterpret_cast<const Provider_NodeArgInfo*>(&reinterpret_cast<const NodeArg*>(p)->ToProto()); }
+  bool Provider_NodeArg__Exists(const Provider_NodeArg* p) const noexcept override { return reinterpret_cast<const NodeArg*>(p)->Exists(); }
+  const ONNX_NAMESPACE::Provider_TypeProto* Provider_NodeArg__TypeAsProto(const Provider_NodeArg* p) noexcept override { return reinterpret_cast<const ONNX_NAMESPACE::Provider_TypeProto*>(reinterpret_cast<const NodeArg*>(p)->TypeAsProto()); }
 
-  Provider_AllocatorPtr Provider_GetAllocator(int id, OrtMemType mem_type) const override {
-    return std::make_shared<Provider_IAllocator_Impl>(IExecutionProvider::GetAllocator(id, mem_type));
-  }
+  // Provider_NodeAttributes
+  std::unique_ptr<Provider_NodeAttributes> Provider_NodeAttributes__construct() override { return std::unique_ptr<Provider_NodeAttributes>(reinterpret_cast<Provider_NodeAttributes*>(new NodeAttributes())); }
+  void Provider_NodeAttributes__operator_delete(Provider_NodeAttributes* p) noexcept override { delete reinterpret_cast<NodeAttributes*>(p); }
+  size_t Provider_NodeAttributes__size(const Provider_NodeAttributes* p) override { return reinterpret_cast<const NodeAttributes*>(p)->size(); }
+  void Provider_NodeAttributes__clear(Provider_NodeAttributes* p) noexcept override { return reinterpret_cast<NodeAttributes*>(p)->clear(); }
+  Provider_AttributeProto& Provider_NodeAttributes__operator_array(Provider_NodeAttributes* p, const std::string& string) override { return *reinterpret_cast<Provider_AttributeProto*>(&(*reinterpret_cast<NodeAttributes*>(p))[string]); }
+  void Provider_NodeAttributes__operator_assign(Provider_NodeAttributes* p, const Provider_NodeAttributes& v) override { *reinterpret_cast<NodeAttributes*>(p) = *reinterpret_cast<const NodeAttributes*>(&v); }
 
-  void Provider_InsertAllocator(Provider_AllocatorPtr allocator) override {
-    IExecutionProvider::InsertAllocator(static_cast<Provider_IAllocator_Impl*>(allocator.get())->p_);
+  std::unique_ptr<Provider_NodeAttributes_Iterator> Provider_NodeAttributes__begin(const Provider_NodeAttributes* p) override {
```

Comment:
> p) o [](start = 113, length = 5)

Should being()/end() be const?

Reply:
The iterator it gets back in this case is only for const iteration. I haven't hit a case that needs both const and non const iteration yet. I figure I'll add multiple types for when that happens. I can rename it to be Provider_NodeAttributes_const_iterator if you prefer that.